src/driver.jl

```diff
@@ -221,8 +222,8 @@ const __llvm_initialized = Ref(false)
             for dyn_job in keys(worklist)
                 # cached compilation
                 dyn_entry_fn = get!(jobs, dyn_job) do
-                    dyn_ir, dyn_meta = codegen(:llvm, dyn_job; toplevel=false,
-                                                               parent_job=job)
+                    config = CompilerConfig(dyn_job.config; toplevel=false)
+                    dyn_ir, dyn_meta = codegen(:llvm, CompilerJob(dyn_job; config))
```

Comment:
I know quite a delayed review, but the removal of `parent_job` here is tricky.

Enzyme places a fully featured `CompilerJob` into the jobs list (https://github.com/EnzymeAD/Enzyme.jl/blob/b9b12e5d6f5597d70bacbe7a83361244253b84bd/src/compiler.jl#L6057) with the target being `EnzymeTarget()`.

Arguably, EnzymeTarget() should probably capture the target of the primal, but at that moment we are executing a generator function (I know I know) and don't have information in which context this job will be used. We somehow need to know the config (or at least target) for the parent such that we could generate the correct primal code, otherwise we default to the CPU target.

We were using `parent_job` to obtain this information just before we went and generated the primal code.